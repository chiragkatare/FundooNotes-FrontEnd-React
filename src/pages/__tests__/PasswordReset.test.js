import React from 'react';
// import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import PasswordReset from '../PasswordReset';
const mockFn = jest.fn();


// configure({ adapter: new Adapter() });

describe('PasswordReset', () => {



  it('should render correctly in "debug" mode', () => {
    const component = shallow(<PasswordReset debug />);
    // const component = mount(<Login debug />);
    expect(component).toMatchSnapshot();
  });

  it('should render correctly in if password token is correct mode', () => {
    const component = shallow(<PasswordReset debug />);
   component.setState({status:true});

 
   expect(component.state('status')).toEqual(true);
  });

  it('password input Should respond to change event and change state of the Login Component', () => {

    const componentInput = shallow(<PasswordReset />);
    const Test = componentInput.find('#password');
  
    Test.props().onChange({
        target: {
            name: 'password',
            value: 'abcd@123'
        }
    });
  
    expect(componentInput.state('password')).toEqual('abcd@123');
  })

  it('confirm password input Should respond to change event and change state of the passwordreset Component', () => {

    const componentInput = shallow(<PasswordReset />);
    const Test = componentInput.find('#rpassword');
  
    Test.props().onChange({
        target: {
            name: 'rpassword',
            value: 'abcd@123'
        }
    });
  
    expect(componentInput.state('rpassword')).toEqual('abcd@123');
  })

});