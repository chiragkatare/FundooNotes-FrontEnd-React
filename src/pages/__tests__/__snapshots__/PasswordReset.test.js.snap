// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordReset should render correctly in "debug" mode 1`] = `
<div>
  <WithStyles(Card)
    id="verify-card"
  >
    <div
      className="label-page"
    >
      <div
        className="logo"
      >
        <img
          alt=""
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "logo.svg",
            }
          }
        />
      </div>
      <div>
        <WithStyles(Typography)
          align="center"
          className="label-text"
          component="h4"
          variant="h5"
        >
          Password Reset
        </WithStyles(Typography)>
        <WithStyles(Typography)
          className="label-text"
          component="p"
        />
      </div>
    </div>
    <WithStyles(CardContent)>
      <div>
         
        <WithStyles(Typography)
          align="center"
          component="h4"
          variant="h5"
        >
          Enter New Password
        </WithStyles(Typography)>
        <div>
          <Input
            id="password"
            label="Password"
            name="password"
            onChange={[Function]}
            placeholder="Enter password"
            type="password"
          />
        </div>
        <div>
          <Input
            id="rpassword"
            label="Confirm Password"
            name="rpassword"
            onChange={[Function]}
            placeholder="Confirm password"
            type="password"
          />
        </div>
        <div
          className="form-btn"
        >
          <WithStyles(Button)
            color="primary"
            fullWidth={true}
            onClick={[Function]}
            type="submit"
            variant="contained"
          >
            Reset
          </WithStyles(Button)>
        </div>
      </div>
    </WithStyles(CardContent)>
  </WithStyles(Card)>
</div>
`;
